docs/resources/sql_endpoint.md

```diff
@@ -36,7 +36,7 @@ The following arguments are supported:
 * `tags` - Databricks tags all endpoint resources with these tags.
 * `spot_instance_policy` - The spot policy to use for allocating instances to clusters: `COST_OPTIMIZED` or `RELIABILITY_OPTIMIZED`. This field is optional. Default is `COST_OPTIMIZED`.
 * `enable_photon` - Whether to enable [Photon](https://databricks.com/product/delta-engine). This field is optional and is enabled by default.
-* `enable_serverless_compute` - Whether this SQL endpoint is a Serverless endpoint. To use a Serverless SQL endpoint, you must enable Serverless SQL endpoints for the workspace.
+* `enable_serverless_compute` - Whether this SQL endpoint is a Serverless endpoint. To use a Serverless SQL endpoint, you must enable Serverless SQL endpoints for the workspace. *Please note that this value has default value of `false` even if Serverless SQL endpoints are enabled for your workspace.* 
```

Comment:
please remind me what's the default in UI?

Reply:
It’s variable - it’s false when it’s disabled for workspace, and true when it’s enabled on workspace level. If you make it true by default and it’s not enabled for workspace then request fails

Reply:
Please set default to true and remove this doc update, otherwise it’s a breaking change for terraform provider and we’re not allowing it. Customers rely on this value being true in production.

Reply:
> if serverless is disabled, setting to true will lead to failure

isn't it what is happening now? If so - this is a Public API.

Terraform resource is _public api of terraform provider_. With v1.x we're no longer breaking the Public API. https://semver.org/ is the source of truth.

Reply:
My point is that we can't make it `default:true` in annotation as it will fail for cases when serverless isn't enabled for workspace

Reply:
it's too big to be rolled out without comms

Reply:
Or we somehow figure out if serverless is enabled or not on the workspace (either via API if we have it, or by handling error), check if the value is explicitly set vs. default...